### ISSUE TITLE
In-class typedef template use

### ISSUE THREAD

**Issue by Pavel-Boyazov** [gajus/eslint-plugin-jsdoc#1655]
## Expected behavior

No errors in console when use template in in-class typedef

## Actual behavior

% eslint .

/myfile.js
  5:1  error  The type 'T' is undefined  jsdoc/no-undefined-types

## ESLint Config

```js
import jsdoc from "eslint-plugin-jsdoc";
import globals from "globals";

export default [
	{
		plugins: { jsdoc },

		languageOptions: {
			globals: {
				...globals.node,
				NodeJS: true,
			},

			sourceType: "commonjs",
		},

		rules: {
			"jsdoc/no-undefined-types": ["error", { markVariablesAsUsed: false }],
		},
	},
];

```

## ESLint sample

```js
// Format JS code here
/**
 * @template T
 */
class Test {
  /** @typedef {T} Type */
  /** @type {Type} */
  t;
}

console.log(Test);
```

## Environment

- Node version: v25.6.0
- ESLint version v9.39.2
- `eslint-plugin-jsdoc` version: 62.5.3
